airflow/serialization/serialized_objects.py

```diff
@@ -228,6 +257,40 @@ def decode_timetable(var: dict[str, Any]) -> Timetable:
     return timetable_class.deserialize(var[Encoding.VAR])
 
 
+def _encode_priority_weight_strategy(var: PriorityWeightStrategy) -> dict[str, Any]:
+    """
+    Encode a priority weight strategy instance.
+
+    This delegates most of the serialization work to the type, so the behavior
+    can be completely controlled by a custom subclass.
+    """
+    priority_weight_strategy_class = type(var)
+    importable_string = qualname(priority_weight_strategy_class)
+    if _get_registered_priority_weight_strategy(importable_string) is None:
+        raise _PriorityWeightStrategyNotRegistered(importable_string)
+    return {Encoding.TYPE: importable_string, Encoding.VAR: var.serialize()}
```

Comment:
Do we assume we need to store a state for the strategy? Would it not enough just to use the task context information and the python code? Would we assume real "data" needs to be serialized for a custom strategy that need to be stored/retrieved? I feel a lot of overhead for very special use cases where strategy state information needs to be persisted.
Do you have a use case in mind where per task (other than the context) data needs to be persisted in a custom strategy? I feel this is a niche use case - but maybe I don't have a use case in mind.